cip/1.accepted/CIP2017-03-01-SKIP-and-LIMIT.adoc

```diff
@@ -0,0 +1,101 @@
+= CIP2017-03-01 - LIMIT subclause
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Mats Rydberg <mats@neotechnology.com>
+
+toc::[]
+
+== Background
+
+This CIP is a proposal in answer to link:https://github.com/opencypher/openCypher/issues/194[CIR-2017-194].
+
+== Proposal
+
+The `LIMIT` subclause is used to constrain the cardinality of its parent clause by providing an upper limit.
+This can be useful for data exploration, or verifying partial results of expensive queries.
+
+=== Syntax
+
+.Syntax overview:
+[source, ebnf]
+----
+clause-with-limit = read-only-clause, [ limit ] ;
+read-only-clause  = match
+                  | with
+                  | unwind
+                  | return
+                  ;
+limit             = "LIMIT", expr ;
+----
+
+=== Semantics
+
+The `LIMIT` subclause prevents records passing through its parent clause after the specified amount of rows, as determined by the limit expression, has been processed.
+For these semantics to be well defined, the limit expression must be constant over the query lifetime, such as parameters or literals.
+
+==== Updating queries
+
+The use of `LIMIT` opens the possibility for certain performance optimisations.
+Clauses that come early in the query do not have to be evaluated over the full dataset, just enough to reach the subsequent limit.
```

Comment:
"do not have to be evaluated over the full dataset, just enough to reach the subsequent limit."
->
"only need to be evaluated until the limit is reached, as opposed to evaluating the entire dataset."